python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -11,11 +12,24 @@ from pylibcudf.libcudf.types cimport size_type
 from rmm.pylibrmm.device_buffer cimport DeviceBuffer
 
 from .gpumemoryview cimport gpumemoryview
+from .filling cimport sequence
 from .scalar cimport Scalar
 from .types cimport DataType, size_of, type_id
 from .utils cimport int_to_bitmask_ptr, int_to_void_ptr
 
-import functools
+from functools import cache
+
+
+try:
+    import numpy as np
+except ImportError:
+    np = None
+
+try:
+    import cupy as cp
+except ImportError:
+    cp = None
```

Comment:
I would prefer for these to be lazily imported so that if we don't need this functionality we don't pay the import cost (importing cupy costs about 3 times as long as importing all of pylibcudf right now).